tests/unit/diddoc/request/diddoc_service_params_test.go

```diff
@@ -0,0 +1,144 @@
+
+	err := didDocService.DidDocEchoHandler(context)
+	if testCase.expectedError != nil {
+		Expect(testCase.expectedError.Error()).To(Equal(err.Error()))
+	} else {
+		Expect(rec.Code).To(Equal(http.StatusSeeOther))
+		Expect(string(testCase.expectedResolution.GetBytes())).To(Equal(context.Response().Header().Get("Location")))
+		Expect(err).To(BeNil())
+	}
+},
+
+	// Positive cases
+	Entry(
+		"Positive. Service case",
+		QueriesDIDDocTestCase{
+			didURL:             fmt.Sprintf("/1.0/identifiers/%s?service=%s", testconstants.ValidDid, testconstants.ValidServiceId),
+			resolutionType:     types.DIDJSONLD,
+			expectedResolution: types.NewServiceResult(testconstants.ValidService.ServiceEndpoint[0]),
+			expectedError:      nil,
+		},
+	),
+	Entry(
+		"Positive. relativeRef case",
+		QueriesDIDDocTestCase{
+			didURL:             fmt.Sprintf("/1.0/identifiers/%s?service=%s&relativeRef=foo", testconstants.ValidDid, testconstants.ValidServiceId),
+			resolutionType:     types.DIDJSONLD,
+			expectedResolution: types.NewServiceResult(testconstants.ValidService.ServiceEndpoint[0] + "foo"),
+			expectedError:      nil,
+		},
+	),
+	Entry(
+		"Positive. VersionId + Service case",
+		QueriesDIDDocTestCase{
+			didURL:             fmt.Sprintf("/1.0/identifiers/%s?versionId=%s&service=%s", testconstants.ValidDid, testconstants.ValidVersionId, testconstants.ValidServiceId),
+			resolutionType:     types.DIDJSONLD,
+			expectedResolution: types.NewServiceResult(testconstants.ValidService.ServiceEndpoint[0]),
+			expectedError:      nil,
+		},
+	),
+	Entry(
+		"Positive. VersionId + Service case + relativeRef",
+		QueriesDIDDocTestCase{
+			didURL:             fmt.Sprintf("/1.0/identifiers/%s?versionId=%s&service=%s&relativeRef=foo", testconstants.ValidDid, testconstants.ValidVersionId, testconstants.ValidServiceId),
+			resolutionType:     types.DIDJSONLD,
+			expectedResolution: types.NewServiceResult(testconstants.ValidService.ServiceEndpoint[0] + "foo"),
+			expectedError:      nil,
+		},
+	),
+	Entry(
+		"Positive. VersionTime + Service case",
+		QueriesDIDDocTestCase{
+			didURL:             fmt.Sprintf("/1.0/identifiers/%s?versionTime=%s&service=%s", testconstants.ValidDid, testconstants.CreatedAfter.Format(time.RFC3339), testconstants.ValidServiceId),
+			resolutionType:     types.DIDJSONLD,
+			expectedResolution: types.NewServiceResult(testconstants.ValidService.ServiceEndpoint[0]),
+			expectedError:      nil,
+		},
+	),
+	Entry(
+		"Positive. VersionTime + Service case + relativeRef",
+		QueriesDIDDocTestCase{
+			didURL:             fmt.Sprintf("/1.0/identifiers/%s?versionTime=%s&service=%s&relativeRef=foo", testconstants.ValidDid, testconstants.CreatedAfter.Format(time.RFC3339), testconstants.ValidServiceId),
+			resolutionType:     types.DIDJSONLD,
+			expectedResolution: types.NewServiceResult(testconstants.ValidService.ServiceEndpoint[0] + "foo"),
+			expectedError:      nil,
+		},
+	),
+
+	// Negative Cases
```

Comment:
Let's add a negative test for testing `relativeRef` query without `service` query.